recommenders/evaluation/spark_evaluation.py

```diff
@@ -630,6 +660,43 @@ def _get_cosine_similarity(self, n_partitions=200):
             )
         return self.df_cosine_similarity
 
+    @staticmethod
+    @udf(returnType=DoubleType())
+    def sim_cos(v1,v2):
+        try:
+            p = 2
+            return float(v1.dot(v2))/float(v1.norm(p)*v2.norm(p))
```

Comment:
Do you know whether using `pyspark.ml.stat.Correlation.corr()` would be faster or slower?